core/src/main/java/io/aiven/kafka/tieredstorage/fetch/cache/DiskChunkCache.java

```diff
@@ -98,11 +98,16 @@ public RemovalListener<ChunkKey, Path> removalListener() {
         return (key, path, cause) -> {
             try {
                 if (path != null) {
-                    final long fileSize = Files.size(path);
-                    Files.delete(path);
-                    metrics.chunkDeleted(fileSize);
-                    log.trace("Deleted cached file for key {} with path {} from cache directory."
-                        + " The reason of the deletion is {}", key, path, cause);
+                    if (Files.exists(path)) {
```

Comment:
I still failing to understand how it is possible that `path` will be null or file will not exist. I think it should not be the case or otherwise something is quite wrong IMO.

Reply:
I agree it's a weird case.

The RemovalListener states: 
> An instance may be called concurrently by multiple threads to process different entries.

This confirms that it could be a race condition by separate threads calling the removal. I also find weird that if there's a wining thread, the metric is not increased.

Also

> Implementations of this interface should avoid performing blocking calls or synchronizing on shared resources.

We currently already handle the case where `path`/value is null by logging an error. This could be seen as an extension of handling the scenario where the referenced file is not found.
We are still logging this to troubleshoot if/when this happens.